### PR TITLE
Compose file: fix variable substitution

### LINKED ISSUE #1933: "docker-compose up" fails.
Running "docker-compose up" fails.

## Details
- OS version: Pop!_OS 18.10
- xterm.js version: 3.11.0
- Docker version: 18.09.0, build 4d60db4
- docker-compose version: 1.22.0, build f46880fe

### Steps to reproduce

1. `docker-compose up`
2. See the error `ERROR: Invalid interpolation format for "ports" option in service "web": "${XTERMJS_PORT:3000}:3000"`

### PR DESCRIPTION
**Pull request by ahtshamraziq**
Closes #1933 
See: https://docs.docker.com/compose/compose-file/#variable-substitution.

### CODE REVIEW

**Comment by Tyriar** [comment]
@ahtshamraziq do you know why this would have broken suddenly? It's been this way for a long time 🤔

**Comment by ahtshamraziq** [comment]
@Tyriar  No clue :sheep:

**Comment by Tyriar** [comment]
I tried running this via `docker-compose up` and I'm getting this error:

```
web_1    | module.js:682
web_1    |   return process.dlopen(module, path._makeLong(filename));
web_1    |                  ^
web_1    | 
web_1    | Error: /usr/src/app/node_modules/node-pty/build/Release/pty.node: invalid ELF header
web_1    |     at Object.Module._extensions..node (module.js:682:18)
web_1    |     at Module.load (module.js:566:32)
web_1    |     at tryModuleLoad (module.js:506:12)
web_1    |     at Function.Module._load (module.js:498:3)
web_1    |     at Module.require (module.js:597:17)
web_1    |     at require (internal/module.js:11:18)
web_1    |     at Object.<anonymous> (/usr/src/app/node_modules/node-pty/lib/unixTerminal.js:22:11)
web_1    |     at Module._compile (module.js:653:30)
web_1    |     at Object.Module._extensions..js (module.js:664:10)
web_1    |     at Module.load (module.js:566:32)
web_1    | npm ERR! code ELIFECYCLE
web_1    | npm ERR! errno 1
web_1    | npm ERR! xterm@3.11.0 start: `node demo/start`
web_1    | npm ERR! Exit status 1
web_1    | npm ERR! 
web_1    | npm ERR! Failed at the xterm@3.11.0 start script.
web_1    | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
web_1    | 
web_1    | npm ERR! A complete log of this run can be found in:
web_1    | npm ERR!     /root/.npm/_logs/2019-03-04T17_54_22_986Z-debug.log
xtermjs_web_1 exited with code 1
```

Any ideas? I haven't tried running this before 😅

**Comment by ahtshamraziq** [comment]
@Tyriar i guess your node version differs from docker base image used here. maybe your `node_modules` folder got mounted into docker environment. using `npm ci` instead of `npm install` in `Dockerflie` could alleviate this.

the whole docker path seems less traveled. it needs an overhaul.

**Comment by ahtshamraziq** [comment]
Here's how I reproduced the issue:
1. Run `yarn`, 
2. Run `docker-compose up`.

To fix it, i deleted `node_modules` directory and ran `docker-compose up` again.
This time both docker containers `web` and `watch` started their own `npm install` concurrently, and failed.

So, I deleted `node_modules` again and ran `docker-compose up web` which succeeded.
Finally, I stopped the `web` contaiiner and ran `docker-compose up` again. This time both containers stayed up.

In summary there are two issues here,
The project root directory gets mounted inside docker containers so if the `node_modules` directory already exists, the above error occurs.
If `node_modules` directory does not exist, both containers run `npm install` simultaneously and fail.
